robottelo/test.py

```diff
@@ -101,6 +102,99 @@
 LOGGER = logging.getLogger(__name__)
 
 
+class _AssertNotRaisesContext(object):
+    """A context manager used to implement :meth:`TestCase.assertNotRaises`.
+    """
+
+    def __init__(self, expected, test_case, expected_regex=None,
+                 cli_return_code=None, http_status_code=None):
```

Comment:
@ldjebran frankly, i'd rather want to not decide by my own, but to hear others thoughts, as this isn't something i'll use exclusively, but a tool for everyone so the option which fits all of us should win instead of one's personal preference :)
But ok, my imo - from design perspective - no doubt, your suggestion is better, from usage perspective - it's far more complicated, user should remember to pass some `value_handler` arg, remember to import one of correct value helpers and pass it as a value. I'd prefer to keep interface as simple as possible, as this assertion itself isn't the simplest one, let's not overcomplicate its usage.